<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1645</width>
    <height>660</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetMinimumSize</enum>
      </property>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="styleSheet">
         <string notr="true">QGroupBox {
    border: 1px solid gray;
    margin-top: 0.5em;
 font-weight:bold
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
 font-weight:bold
   
}</string>
        </property>
        <property name="title">
         <string>Camera settings</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="5" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QDoubleSpinBox" name="roiXInput">
             <property name="suffix">
              <string> px</string>
             </property>
             <property name="decimals">
              <number>0</number>
             </property>
             <property name="maximum">
              <double>999999999.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="roiYInput">
             <property name="suffix">
              <string> px</string>
             </property>
             <property name="decimals">
              <number>0</number>
             </property>
             <property name="maximum">
              <double>999999999.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="3" column="1">
          <widget class="QComboBox" name="clearModeSelect"/>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_4">
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="text">
            <string>Clear Mode:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="text">
            <string>Exposure Time:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QDoubleSpinBox" name="roiXOffsetInput">
             <property name="suffix">
              <string> px</string>
             </property>
             <property name="decimals">
              <number>0</number>
             </property>
             <property name="maximum">
              <double>999999999.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="roiYOffsetInput">
             <property name="suffix">
              <string> px</string>
             </property>
             <property name="decimals">
              <number>0</number>
             </property>
             <property name="maximum">
              <double>999999999.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="label_6">
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="text">
            <string>Roi offset:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="readoutRateSelect"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="text">
            <string>Readout Rate</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QPushButton" name="resetRoiButton">
             <property name="text">
              <string>Reset ROI</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="setRoiButton">
             <property name="text">
              <string>Set ROI</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_5">
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="text">
            <string>Roi size:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="exposureInput">
           <property name="suffix">
            <string> ms</string>
           </property>
           <property name="maximum">
            <double>999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="text">
            <string>Trigger Mode:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="triggerModeSelect"/>
         </item>
         <item row="4" column="0" colspan="2">
          <widget class="Line" name="line_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="0" colspan="3">
       <widget class="QFrame" name="cameraViewFrame">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>1</verstretch>
         </sizepolicy>
        </property>
        <property name="frameShape">
         <enum>QFrame::Box</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_4"/>
       </widget>
      </item>
      <item row="0" column="0" colspan="3">
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QPushButton" name="liveCameraButton">
          <property name="text">
           <string>Live</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="snapFrameButton">
          <property name="text">
           <string>Snap</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QSpinBox" name="cameraFramesInput">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="cameraFramesRecordButton">
            <property name="text">
             <string>Record</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="bgSubstracionFramesInput">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="cameraBgSubstractionButton">
            <property name="text">
             <string>Bg. Substraction</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="1" column="0">
       <widget class="QFrame" name="graphViewFrame">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>1</verstretch>
         </sizepolicy>
        </property>
        <property name="frameShape">
         <enum>QFrame::Box</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_5"/>
       </widget>
      </item>
      <item row="0" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QPushButton" name="startButton">
          <property name="text">
           <string>Start</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="refreshButton">
          <property name="text">
           <string>Refresh</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_11">
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="saveGraphButton">
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="graphFramesRecordInput">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="recordGraphButton">
            <property name="text">
             <string>Record</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="saveImagesCheckbox">
            <property name="text">
             <string>Save images</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1645</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuApplication">
    <property name="title">
     <string>Application</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuApplication"/>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>resetRoiButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>resetRoi()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setRoiButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>setRoi()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>368</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearModeSelect</sender>
   <signal>editTextChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>clearModeChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>174</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>187</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>triggerModeSelect</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>triggerModeChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>152</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>readoutRateSelect</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>readoutRateChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>122</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exposureInput</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>exposureTimeChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>278</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>347</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>liveCameraButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>liveModeClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>422</x>
     <y>47</y>
    </hint>
    <hint type="destinationlabel">
     <x>463</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>snapFrameButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>snapClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>511</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>789</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraFramesRecordButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>cameraRecordClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>729</x>
     <y>51</y>
    </hint>
    <hint type="destinationlabel">
     <x>377</x>
     <y>659</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraBgSubstractionButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>bgSubstracionClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>929</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>659</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>refreshButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1141</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>1143</x>
     <y>-130</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveGraphButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>graphSaveClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1310</x>
     <y>52</y>
    </hint>
    <hint type="destinationlabel">
     <x>1304</x>
     <y>-80</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>recordGraphButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>recordGraphClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1470</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>1458</x>
     <y>-42</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>822</x>
     <y>329</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1097</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>1080</x>
     <y>-81</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>resetRoi()</slot>
  <slot>setRoi()</slot>
  <slot>clearModeChanged(QString)</slot>
  <slot>triggerModeChanged(QString)</slot>
  <slot>readoutRateChanged(QString)</slot>
  <slot>exposureTimeChanged(double)</slot>
  <slot>liveModeClicked()</slot>
  <slot>snapClicked()</slot>
  <slot>cameraRecordClicked()</slot>
  <slot>bgSubstracionClicked()</slot>
  <slot>startClicked()</slot>
  <slot>refreshClicked()</slot>
  <slot>graphSaveClicked()</slot>
  <slot>recordGraphClicked()</slot>
  <slot>start()</slot>
 </slots>
</ui>
